app/models/job_execution.rb

```diff
@@ -11,33 +11,43 @@ class JobExecution
   cattr_accessor(:lock_timeout, instance_writer: false) { 10.minutes }
   cattr_accessor(:stop_timeout, instance_writer: false) { 15.seconds }
 
-  cattr_reader(:registry, instance_accessor: false) { {} }
+  cattr_reader(:registry, instance_accessor: false) { JobQueue.new }
   private_class_method :registry
 
   attr_reader :output, :job, :viewers, :stage, :executor
 
-  def initialize(reference, job)
+  def initialize(reference, job, **env, &block)
```

Comment:
Why a double-splat for env?  Shouldn't it just be a regular hash that gets passed in?